skbio/stats/composition.py

```diff
@@ -1105,6 +1227,64 @@ def _gram_schmidt_basis(n):
     return basis.T
 
 
+def sbp_basis(sbp):
+    """
+    Builds an orthogonal basis from a sequential binary partition (SBP).
+    As explained in Parent et al. (2013), the "SBP describes the D−1
+    orthogonal (geometrically independent) balances between parts and
+    groups of parts. The SBP is a (D - 1) × D matrix, in which parts
+    labeled '+1' (group  numerator) are  balanced  with  parts  labeled
+    '−1' (group denominator). A part labeled '0' is excluded from the
+    balance between parts. The composition is partitioned sequentially
+    into contrasts at every hierarchically ordered row until the '+1'
+    and '-1' groups each contain a single part."
+
+    Parameters
+    ----------
+    sbp : numpy.array of integers
+          1, -1 and 0
+
+    Returns
+    -------
+    numpy.array
+      Am orthonormal basis in the Aitchison simplex
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> sbp = np.array([[1, 1,-1,-1,-1],
+    ...                 [1,-1, 0, 0, 0],
+    ...                 [0, 0, 1,-1,-1],
+    ...                 [0, 0, 0, 1,-1]])
+    ...
+    >>> sbp_basis(sbp)
+    array([[ 0.31209907,  0.31209907,  0.12526729,  0.12526729,  0.12526729],
+           [ 0.36733337,  0.08930489,  0.18112058,  0.18112058,  0.18112058],
+           [ 0.17882092,  0.17882092,  0.40459293,  0.11888261,  0.11888261],
+           [ 0.18112058,  0.18112058,  0.18112058,  0.36733337,  0.08930489]])
+
+    References
+    ----------
+    .. [1] Parent, S.É., Parent, L.E., Egozcue, J.J., Rozane, D.E.,
+       Hernandes, A., Lapointe, L., Hébert-Gentile, V., Naess, K.,
+       Marchand, S., Lafond, J., Mattos, D., Barlow, P., Natale, W., 2013.
+       The plant ionome revisited by the nutrient balance concept.
+       Front. Plant Sci. 4, 39, http://dx.doi.org/10.3389/fpls.2013.00039.
+    """
+
+    dim_sbp = sbp.shape
+    isPos = (sbp > 0)
+    isNeg = (sbp < 0)
+    onesD = np.ones((dim_sbp[1], dim_sbp[1]))
+    nPos = np.dot(isPos, onesD)
+    nNeg = np.dot(isNeg, onesD)
+    W = (isPos * nNeg - isNeg * nPos)
+    nn = np.apply_along_axis(lambda x: 1/np.sqrt(np.dot(x, x)), 1, W)
+    nn = np.array([nn, ] * dim_sbp[1]).T
```

Comment:
We'll probably want to replace
```python
nn = np.apply_along_axis(lambda x: 1/np.sqrt(np.dot(x, x)), 1, W)
nn = np.array([nn, ] * dim_sbp[1]).T
```
with
```python
    def _f(x):  return 1 / np.sqrt(np.dot(x, x))    
    nn = np.vstack([_f(W[i, :]) for i in range(W.shape[0])])
```

This is because numpy has changed the return syntax in the newest release.  So we are avoiding using `apply_along_axis` for future releases (see this PR: https://github.com/biocore/scikit-bio/pull/1472)